Sources/Datadog/RUM/RUMMonitor/Scopes/RUMScopeDependencies.swift

```diff
@@ -73,7 +71,51 @@ internal extension RUMScopeDependencies {
             vitalCPUReader: rumFeature.vitalCPUReader,
             vitalMemoryReader: rumFeature.vitalMemoryReader,
             vitalRefreshRateReader: rumFeature.vitalRefreshRateReader,
-            onSessionStart: rumFeature.onSessionStart
+            sessionsListeners: rumFeature.sessionsListeners
         )
     }
 }
+
+/// Objects complying to `RUMSessionListener` will be able to receive
+/// RUM session start notifications from `RUMSessionListeners` after
+/// registration.
+internal protocol RUMSessionListener {
+    /// Receives RUM sessions start information.
+    /// - Parameters:
+    ///   - id: The Session UUID
+    ///   - isDiscarded: `true` if the session is discarded by sampling.
+    func onSessionStart(session id: String, isDiscarded: Bool)
+}
+
+/// `RUMSessionListeners` support registration of session start listeners.
+internal final class RUMSessionListeners {
+    typealias Listener = (String, Bool) -> Void
+
+    private var listeners: [Listener] = []
```

Comment:
We access `listeners` array from multiple threads, including:
* `sessionsListeners.register()` in main `Datadog` class - called on the user thread,
* `sessionsListeners.notify()` from `RUMApplicationScope` - called on RUM's thread,

so it should be synchronised 💥. For what I see, now all `register()` calls happen from `Datadog.initialize()` and running into `notify()` flow requires the SDK to be initialized, so this can't crash for now. However, this assumption can be easily broken if calling one more `register()` in after SDK init.

Alternatively, we could make `listeners` immutable by requiring it in `RUMSessionListeners.init()`, but this option doesn't look easier.

Reply:
Oh indeed! I would like to avoid using a queue for this. I will see if I can achieve immutability of `RUMSessionListeners`, otherwise we could so `objc_sync_enter/objc_sync_exit`.